process/transactions/tokens.go

```diff
@@ -0,0 +1,236 @@
+	if tp.selfShardID != core.MetachainShardId {
+		return []*data.TokenInfo{}
+	}
+
+	tokensData := make([]*data.TokenInfo, 0)
+	for _, scr := range scrs {
+		funcName, resScr, ok := tp.isIssueSCRSuccess(scr.Data, scr.Sender, scrs)
+		if !ok {
+			continue
+		}
+
+		tokenInfo, ok := tp.extractTokenInfo(scr.Data, resScr)
+		if !ok {
+			continue
+		}
+
+		if funcName != issueFungibleESDTFunc {
+			tokensData = append(tokensData, tokenInfo)
+			continue
+		}
+
+		identifier := tp.searchTokenIdentifierFungibleESDT(scr.Sender, scrs)
+		if identifier == "" {
+			continue
+		}
+		tokenInfo.Issuer = scr.Sender
+		tokenInfo.Identifier = identifier
+
+		tokensData = append(tokensData, tokenInfo)
+	}
+
+	return tokensData
+}
+
+func (tp *tokensProcessor) searchTokenIdentifierFungibleESDT(sender string, scrs []*data.ScResult) string {
+	for _, scr := range scrs {
+		isReceiverAndSenderOk := scr.Receiver == sender && scr.Sender == metachainESDTAddress
+		if !isReceiverAndSenderOk {
+			continue
+		}
+		if scr.Nonce != 0 {
+			continue
+		}
+
+		funcName, args, err := tp.argumentParserExtended.ParseData(string(scr.Data))
+		if err != nil {
+			continue
+		}
+
+		if funcName != core.BuiltInFunctionESDTTransfer || len(args) < 2 {
+			continue
+		}
+
+		return string(args[0])
+	}
+
+	return ""
+}
+
+func (tp *tokensProcessor) extractTokenInfo(dataField []byte, scr *data.ScResult) (*data.TokenInfo, bool) {
+	funcName, args, err := tp.argumentParserExtended.ParseData(string(dataField))
+	if err != nil {
+		return nil, false
+	}
+
+	if len(args) < 2 {
+		return nil, false
+	}
+
+	tokenInfo := &data.TokenInfo{
+		Name:   string(args[0]),
+		Ticker: string(args[1]),
+		Type:   computeTokenTypeByIssueFunction(funcName),
+	}
+
+	if scr == nil {
+		return tokenInfo, true
+	}
+
+	tokenInfo.Issuer = scr.Receiver
+
+	split := tp.argumentParserExtended.split(string(scr.Data))
+	if len(split) < 3 {
+		return tokenInfo, true
+	}
+
+	tokenIdentifier, err := hex.DecodeString(split[2])
+	if err != nil {
+		return nil, false
+	}
+
+	tokenInfo.Identifier = string(tokenIdentifier)
+
+	return tokenInfo, true
+}
+
+func (tp *tokensProcessor) isIssueTxSuccess(dataField []byte, sender string, nonce uint64, scrs []*data.ScResult) (string, *data.ScResult, bool) {
+	funcName, ok := tp.isFuncNameOK(dataField)
+	if !ok {
+		return "", nil, false
+	}
+
+	for _, scr := range scrs {
+		isReceiverAndSenderOk := scr.Receiver == sender && scr.Sender == metachainESDTAddress
+		if !isReceiverAndSenderOk {
+			continue
+		}
+
+		isNonceOk := scr.Nonce == nonce+1
+		if !isNonceOk {
+			continue
+		}
+
+		if tp.argumentParserExtended.hasOKPrefix(string(scr.Data)) {
+			return funcName, scr, true
+		}
+	}
+
+	return "", nil, false
+}
+
+func (tp *tokensProcessor) isFuncNameOK(dataField []byte) (string, bool) {
+	funcName, _, err := tp.argumentParserExtended.ParseData(string(dataField))
+	if err != nil {
+		return "", false
+	}
+
+	_, ok := tp.issueMethods[funcName]
```

Comment:
could have been `return tp.issueMethods[funcName]`

Reply:
is this case I cannot.